src/final/SPI_Camera.cpp

```diff
@@ -4,9 +4,93 @@ SPI_Camera::SPI_Camera()
     : SPI(SPI2) {}
 
 void SPI_Camera::configure_GPIO() {
-	// [TODO]
+	// [DONE]
+	RCC->AHB2ENR |= RCC_AHB2ENR_GPIOBEN; // enable GPIOB
+	// set to alternate function mode (10)
+	// explanation: default is 11, so use mask &~(01) to set second bit to 0
+	GPIOB->MODER &= ~(GPIO_MODER_MODER13_0 | // PB13
+					  GPIO_MODER_MODER14_0 | // PB14
+					  GPIO_MODER_MODER15_0); // PB15
+	
+	// reset AF for PB13, PB14, PB15
+	GPIOB->AFR[1] &= ~(GPIO_AFRH_AFSEL13 | // PB13
+	                   GPIO_AFRH_AFSEL14 | // PB14
+	                   GPIO_AFRH_AFSEL15); // PB15
+
+	// set PB13, PB14, PB15 to AF15
+	af_num = 5;
+	GPIOB->AFR[1] |= (af_num << (13*4)); // PB13
+	GPIOB->AFR[1] |= (af_num << (14*4)); // PB14
+	GPIOB->AFR[1] |= (af_num << (15*4)); // PB15
```

Comment:
```suggestion
	GPIOB->AFR[1] |= (af_num << (13%8*4)); // PB13
	GPIOB->AFR[1] |= (af_num << (14%8*4)); // PB14
	GPIOB->AFR[1] |= (af_num << (15%8*4)); // PB15
```
You're shifting by more than the size of the register.

Each AFR register corresponds to 8 pin positions (4 bits per pin * 8 pins), so the 13th pin actually maps to the 5th position, 14th to 6th, 15th to 7th.